### PR TITLE
Use k8s_info module instead of deprecated k8s_facts in molecule scaffold (#2168)

### LINKED ISSUE #2168: Switch Ansible test scaffold from k8s_facts to k8s_info module
## Feature Request

**Is your feature request related to a problem? Please describe.**

When using the molecule test scaffold shipped with operator-sdk's Ansible option, playbook runs on Ansible 2.9 show a deprecation warning like the following:

```
    TASK [Wait 60s for reconciliation to run] ***************************************
[DEPRECATION WARNING]: The 'k8s_facts' module has been renamed to 'k8s_info'. 
This feature will be removed in version 2.13. Deprecation warnings can be 
disabled by setting deprecation_warnings=False in ansible.cfg.
    ok: [localhost]

```

**Describe the solution you'd like**

This is pretty easy to fix; we just need to change from using `k8s_facts` to `k8s_info`. I verified with a default test setup that this works identically and drops the deprecation warnings.

### PR DESCRIPTION
**Pull request by geerlingguy**
**Description of the change:**
Replaces uses of the deprecated `k8s_facts` module with `k8s_info`, which is the standard module for k8s information lookups in Ansible 2.9+.

**Motivation for the change:**

Closes #2168

### CODE REVIEW

**Comment by fabianvf** [comment]
Do you mind making sure that the various Dockerfiles (in `ci/dockerfiles` as well as `internal/scaffold/ansible/dockerfilehybrid.go`) are explicitly set to pull in >= 2.9?

**Comment by geerlingguy** [comment]
@fabianvf - ah, completely forgot about that. Bumped minimum to 2.9 instead of 2.8, and I think I covered the only three places that's required.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 ### Changed
 - Upgrade minimal Ansible version in the init projects from `2.4` to `2.6`. ([#2107](https://github.com/operator-framework/operator-sdk/pull/2107))
+- Use `k8s_info` module instead of deprecated `k8s_facts` module in molecule test scaffold. ([#2168](https://github.com/operator-framework/operator-sdk/issues/2168))
```
Comment (camilamacedo86) [inline CHANGELOG.md]:
```suggestion
- Replace in the ansible based operators module tests the `k8s_facts` for `k8s_info` which is deprecated. ([#2168](https://github.com/operator-framework/operator-sdk/issues/2168))
```

**Comment by geerlingguy** [comment]
@camilamacedo86 - Made those changes, thanks for pointing out the missing Changelog entry too!

**Comment by geerlingguy** [comment]
Rebased due to CHANGELOG conflicts, as well as some conflicts due to adding `jmespath` to the Dockerfiles in #2027

File: internal/scaffold/ansible/dockerfilehybrid.go [hunk 2]
```diff
@@ -70,8 +70,9 @@ RUN yum clean all && rm -rf /var/cache/yum/* \
       ansible-runner==1.3.4 \
       ansible-runner-http==1.0.0 \
       openshift==0.8.9 \
-      ansible~=2.8 \
+      ansible~=2.9 \
       jmespath \
+>>>>>>> Update Ansible minimum version to 2.9 in various Dockerfiles (#2203)
```
Comment (camilamacedo86) [inline internal/scaffold/ansible/dockerfilehybrid.go]:
```suggestion
```
Ci broken because of ^

**Comment by geerlingguy** [comment]
@camilamacedo86 - Yikes, don't know how that made it through! I have fixed it, sorry about that.

File: internal/scaffold/ansible/dockerfilehybrid.go (hunk 2 above)
Comment (geerlingguy) [reply internal/scaffold/ansible/dockerfilehybrid.go]: Fixed.

**Pull request merged**